nixos/modules/services/networking/openconnect.nix

```
@@ -0,0 +1,154 @@
+{ config, lib, options, pkgs, ... }: with lib;
+let
+  cfg = config.networking.openconnect;
+  openconnect = cfg.package;
+  interfaceOptions = {
+    options = {
+      gateway = mkOption {
+        description = "Gateway server to connect to.";
+        example = "gateway.example.com";
+        type = types.str;
+      };
+
+      protocol = mkOption {
+        description = ''
+          Protocol to use. One of <literal>anyconnect</literal>,
+          <literal>array</literal>, <literal>nc</literal>,
+          <literal>pulse</literal>, <literal>gp</literal>,
+          <literal>f5</literal> and <literal>fortinet</literal>
+        '';
+        example = "anyconnect";
+        type = types.str;
+      };
+
+      user = mkOption {
+        description = "Username to authenticate with.";
+        example = "example-user";
+        type = types.nullOr types.str;
+      };
+
+      # Note: It does not make sense to provide a way to declaratively
+      # set an authentication cookie, because they have to be requested
+      # for every new connection and would only work once.
+      password = mkOption {
+        description = ''
+          Password to authenticate with. WARNING: To prevent this from being
+          stored world-readable in the Nix store, use
+          <code>passwordFile</code> instead. This is passed to
+          <code>openconnect</code> via the <code>--passwd-on-stdin</code> option.
+        '';
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      # Note
+      passwordFile = mkOption {
+        description = ''
+          File containing the password to authenticate with. This
+          is passed to <code>openconnect</code> via the
+          <code>--passwd-on-stdin</code> option.
+        '';
+        default = null;
+        example = "/var/lib/secrets/openconnect-passwd";
+        type = types.path;
```

```suggestion
        type = types.nullOr types.path;
```
otherwise the default value would cause validation failures